perf/change_point_detector.go

```diff
@@ -0,0 +1,79 @@
+package perf
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/evergreen-ci/cedar/util"
+	"github.com/evergreen-ci/gimlet"
+	"github.com/pkg/errors"
+	"net/http"
+)
+
+type ChangeDetector interface {
+	DetectChanges([]float64) ([]ChangePoint, error)
+}
+
+type ChangePoint struct {
+	Index     int
+	Algorithm Algorithm
+}
+
+type Algorithm struct {
+	Name          string
+	Version       string
+	Configuration map[string]float64
+}
+
+type signalProcessingClient struct {
+	token   string
+	baseURL string
+}
+
+func NewChangeDetector(baseURL, token string) ChangeDetector {
```

Comment:
can this have a slightly less generic name, given that it can only construct a single type of detector?